Python/msrd.py

```diff
@@ -29,48 +37,149 @@ def __init__(self, msrd_url, account_id, api_token):
 
         self.session = requests.Session()
         self.session.headers.update(self.headers)
-        self.session.hooks.update({
-            'response': lambda r, *args, **kwargs: print('{} {} {}'.format(r.status_code, r.reason, r.url))
-        })
+        self.session.hooks.update(
+            {
+                'response': lambda r, *args, **kwargs: print(
+                    '{} {} {}'.format(
+                        r.status_code,
+                        r.reason,
+                        r.url
+                    )
+                )
+            }
+        )
 
     def _url(self, fmt, *args, **kwargs):
+        """
+        _url: Format URL.
+        """
         path = fmt.format(*args, **kwargs)
         return urljoin(self.msrd_url, path)
 
     def account_info(self):
+        """
+        Get Account Information via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}', self.account_id)
         return self.session.get(url)
 
     def os_images(self):
+        """
+        Get Available OS Images for the Account via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}/osimages', self.account_id)
         return self.session.get(url)
 
     def job_tiers(self):
+        """
+        Get Available Job Tiers for the Account via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}/jobtiers', self.account_id)
         return self.session.get(url)
 
     def jobs(self):
+        """
+        Get Available Job for the Account via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}/jobs', self.account_id)
         return self.session.get(url)
 
     def upload_file(self, file_path):
+        """
+        Upload file via a REST Call to the MSRD File PUT API.
+        """
         url = self._url('files/accounts/{}/session', self.account_id)
-        with open(file_path, 'rb') as f:
-            return self.session.put(url, data=f)
+        with open(file_path, 'rb') as file_to_upload:
+            return self.session.put(url, data=file_to_upload)
 
     def submit_job(self, job):
+        """
+        Submit Job via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}/jobs', self.account_id)
         return self.session.post(url, json=job)
 
 
-def print_response(r):
+def print_response(response):
+    """
+    Pretty-Print response data
+    """
     try:
-        print(json.dumps(r.json(), indent=2))
+        print(json.dumps(response.json(), indent=2))
     except json.JSONDecodeError:
-        print(r.text)
+        print(response.text)
+
+
+DEFAULT_MSRD_URL = 'https://microsoftsecurityriskdetection.com'
 
+MAX_FILE_SIZE = 4194304  # 4 Megabytes.
+
+
+def generate_file_info(client, file):
```

Comment:
This function does more than generate a file info object: it _uploads_ the file first. So, how about we rename it to `upload_file()`, and then have it return the dict suggested below (a file info object).

Reply:
It's already using `upload_file` for the `upload_file` command.  I'm happy to rename it, however.

Reply:
Renamed to upload_file_and_generate_file_info